graphql/mutation_resolver.go

```diff
@@ -626,6 +626,9 @@ func (r *mutationResolver) EditSpawnHost(ctx context.Context, spawnHost *EditSpa
 		opts.AttachVolume = *spawnHost.Volume
 	}
 	if spawnHost.PublicKey != nil {
+		if h.Status != evergreen.HostRunning {
+			return nil, InputValidationError.Send(ctx, "Host must be running to add a public key")
```

Comment:
You should also add some front-end validation for this. Otherwise, the request will just fail and it will be confusing as to why. We should prevent the interaction from happening altogether.

Reply:
Won't users see this error?

Reply:
We should disable the option to change the ssh key if the host is not running allowing a user to perform an operation that will always fail is a bad experience.

Reply:
Yeah that sounds good. We should also make sure this mouseover (that exists for instance type) shows up. 
<img width="245" alt="image" src="https://user-images.githubusercontent.com/26798134/203092199-7af05b5c-a713-4905-bb3f-f176bab2662f.png">

If that seems too large @bynn feel free to cut a new ticket; I think making sure the users know about the error is priority #1.

Reply:
https://jira.mongodb.org/browse/EVG-18003

I created a ticket for the frontend part of the validation!